test/SlowTests/Issues/RavenDB-17566.cs

```diff
@@ -0,0 +1,116 @@
+﻿using Raven.Client.Documents;
+using Raven.Client.Documents.Indexes;
+using Raven.Client.Documents.Queries;
+using Raven.Client.Documents.Session;
+using System.Linq;
+using System.Threading.Tasks;
+using FastTests;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Issues;
+
+public class RavenDB_17566 : RavenTestBase
+{
+    public RavenDB_17566(ITestOutputHelper output) : base(output)
+    {
+    }
+
+    [Fact]
+    public async Task ProjectionOnNonStoredFieldsInIndex()
+    {
+        using var store = base.GetDocumentStore(new Options
+        {
+            ModifyDocumentStore = s => s.Conventions.FindProjectedPropertyNameForIndex =
```

Comment:
Note: In order to make the tests pass we need to override this convention. In the index definition we don't store the nested field `Details.Description` so we want to look for it from the document when doing the projection. While our default convention would translate it to `Details_Description` and such field doesn't exist in the document.